Spotify Module + Interrupt Jarvis Feature

Supereiris opened the pull request:
Jarvis can now use Spotify!
    - Play a song, playlist or artist's top songs
    - Play a playlist the user has created
    - Option to shuffle playlists
    - Add a song to the queue
    - Shuffle queue
    - Pause/Resume/Skip player
    - Let Jarvis choose and play a song

You can also interrupt Jarvis by pressing the key to talk. Can be toggled off in config.ini

BEAWARE: I have not tested the module with a non-premium Spotify account.

Supereiris committed:
Interrupt Jarvis Feature

When pressing key to talk, Jarvis stops talking

Supereiris committed:
Update config.ini

Supereiris committed:
Update config.py

Supereiris committed:
Spotify Module

Jarvis can now connect to Spotify and use various features.
Refer to CHANGELOG.MD

Supereiris committed:
Merge branch 'main' of https://github.com/Supereiris/Jarvis-Modular-Mark

ifBars commented:
Awesome! Same thing as last time, make sure you sync your fork since I added a few new things that your fork is behind on, and if it has conflicts I can fix them again. Once we get that sorted, I will do a quick test run to make sure the sync didn't break anything, and then we should be good to merge!

EDIT: Spotify module seems to have some issues caused by the spotify app being in developer mode, requiring users to be manually whitelisted with a limit of 25. Option 1 is pushing spotify app to production, which is currently in progress. Option 2 is having users manually create their own spotify apps.

Supereiris committed:
Merge branch 'main' into main

ifBars committed:
Fix spotify issues & add client id to config

Supereiris committed:
Merge pull request #4 from ifBars/main

Fix interrupt_jarvis config issue & add client id to config

Supereiris committed:
Toggleable Modules

Modules can now be enabled/disabled from config.ini

Supereiris committed:
Added Spotify Module Set Up Tutorial

ifBars committed:
Minor Fixes

- Added screen size grabbing to insta lock, as this was lost in the last PR's merge conflicts due to an oversight on my part
- Added enabled_modules to the default config inside config.py, so it gets added to the ini file if need be.

ifBars reviewed (commented):
Made a [PR](https://github.com/Supereiris/Jarvis-Modular-Mark/pull/5) on your repo to add enabled_modules to default_config, and fix an issue I overlooked in the merge conflicts of your first PR. Once that PR is merged, this PR is all set to be merged. Thanks for your continued contributions, I personally really enjoy the Spotify module!

Supereiris committed:
Merge pull request #5 from ifBars/main

Minor Fixes
